common/fetch_keys.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+package common
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/coreos/rkt/pkg/keystore"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/discovery"
+	"github.com/coreos/rkt/Godeps/_workspace/src/golang.org/x/crypto/openpgp"
+)
+
+// GetPubKeyLocation either returns the location supplied in argv or discovers one @ prefix
+func GetPubKeyLocations(prefix string, args []string, allowHTTP bool, debug bool) ([]string, error) {
+	if len(args) > 0 {
+		return args, nil
+	}
+
+	if prefix == "" {
+		return nil, fmt.Errorf("at least one key or --prefix required")
```

Comment:
Part of the reason it made more sense for this to live in package main is because of the coupling to the argument names - e.g "--prefix" here, "--insecure-allow-http" elsewhere. Would be nice to clean this up but it escapes me off the top of my head of how it could be structured.